Kerdy/Kerdy/Common/ModalVC.swift

```diff
@@ -0,0 +1,212 @@
+//
+//  ModalVC.swift
+//  Kerdy
+//
+//  Created by JEONGEUN KIM on 1/11/24.
+//
+
+import UIKit
+
+import Core
+
+import RxSwift
+import RxDataSources
+
+final class ModalVC: UIViewController {
+    
+    // MARK: - Property
+    
+    typealias DataSource = RxCollectionViewSectionedReloadDataSource<CommonModalSectionItem.Model>
+    
+    private var dataSource: DataSource!
+    private let viewModel: CommonModalViewModel
+    private let disposeBag = DisposeBag()
+    private var type: AlertType?
+    
+    // MARK: - UI Components
+    
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: self.layout())
+        collectionView.bounces = false
+        collectionView.makeCornerRound(radius: 15)
+        collectionView.layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.isScrollEnabled = false
+        return collectionView
+    }()
+    
+    private let popupView =  PopUpView()
+    
+    // MARK: - init
+    
+    init(viewModel: CommonModalViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    // MARK: - Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setRegisteration()
+        setLayout()
+        setUI()
+        setDataSource()
+        setDelegate()
+        setBindings()
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesBegan(touches, with: event)
+        let touch = touches.first!
+        let location = touch.location(in: self.view)
+        
+        if !collectionView.frame.contains(location) {
+            view.backgroundColor = .clear
+            dismiss(animated: true)
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Methods
+
+extension ModalVC {
+    
+    private func setRegisteration() {
+        
+        collectionView.register(CommonModalCell.self, forCellWithReuseIdentifier: CommonModalCell.identifier)
+    }
+    
+    private func setLayout() {
+        
+        view.addSubview(collectionView)
+        collectionView.snp.makeConstraints {
+            $0.height.equalTo(120)
+            $0.bottom.horizontalEdges.equalToSuperview()
+        }
+    }
+    
+    private func setUI() {
+        
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.25) {
+            UIView.animate(withDuration: 0.5) {
+                self.view.backgroundColor = .kerdyBlack.withAlphaComponent(0.6)
+            }
+        }
```

Comment:
```suggestion
UIView.animate(
            withDuration: 0.5,
            delay: 0.25
            ) {
                self.view.backgroundColor = .kerdyBlack.withAlphaComponent(0.6)
            }
```
UIView.animate에서 delay를 줌으로서 같은 효과를 얻을 수 있습니다! 참고해주시면 좋을거 같아요!

Reply:
2406a103307fe214838f3ab4acbcc0808487309d